cmd/config-encrypted.go

```diff
@@ -146,44 +153,54 @@ func migrateConfigPrefixToEncrypted(objAPI ObjectLayer, encrypted bool) error {
 		logger.Info(fmt.Sprintf("Attempting to re-encrypt config, IAM users and policies on MinIO with %q (%s)", stat.DefaultKey, stat.Name))
 	}
 
-	var marker string
-	for {
-		res, err := objAPI.ListObjects(GlobalContext, minioMetaBucket, minioConfigPrefix, marker, "", maxObjectList)
+	results := make(chan ObjectInfo)
+	if err := objAPI.Walk(GlobalContext, minioMetaBucket, minioConfigPrefix, results, ObjectOptions{}); err != nil {
+		return err
+	}
+
+	for obj := range results {
+		data, err := readConfig(GlobalContext, objAPI, obj.Name)
 		if err != nil {
 			return err
 		}
-		for _, obj := range res.Objects {
-			data, err := readConfig(GlobalContext, objAPI, obj.Name)
-			if err != nil {
-				return err
-			}
 
-			if !utf8.Valid(data) {
-				data, err = madmin.DecryptData(globalActiveCred.String(), bytes.NewReader(data))
-				if err != nil {
-					return fmt.Errorf("Decrypting config failed %w, possibly credentials are incorrect", err)
-				}
-			}
-			if GlobalKMS != nil {
-				data, err = config.EncryptBytes(GlobalKMS, data, kms.Context{
-					obj.Bucket: path.Join(obj.Bucket, obj.Name),
-				})
-				if err != nil {
-					return err
+		if !utf8.Valid(data) {
+			pdata, err := madmin.DecryptData(globalActiveCred.String(), bytes.NewReader(data))
+			if err != nil {
+				if GlobalKMS != nil {
```

Comment:
Seems to missing the case where `err != nil && GlobalKMS == nil`? This condition appears to be not-handled in `migrateIAMConfigsEtcdToEncrypted` too.

Reply:
well there is nothing we can do at that point if content cannot be decrypted @donatello - so its not handled. what are you thinking that we should do?

Reply:
In the other case when decryption fails, we do `return fmt.Errorf("Decrypting IAM config failed %w, possibly credentials are incorrect", err)` - we can do the same here?

Reply:
Otherwise looks like we are saving/re-encrypting some garbage output.

Reply:
Let me add that perhaps this has been a cause of multiple issues in the past.